WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/usecases/FetchFixThreatsStatusUseCaseTest.kt

```diff
@@ -0,0 +1,123 @@
+        fakeSiteId,
+        listOf(fakeFixThreatsStatusModel),
+        FETCH_FIX_THREATS_STATUS
+    )
+
+    private val storeResultWithFixedFixStatusModel = storeResultWithInProgressFixStatusModel.copy(
+        fixThreatStatusModels = listOf(fakeFixThreatsStatusModel.copy(status = FixStatus.FIXED))
+    )
+
+    @Before
+    fun setup() {
+        useCase = FetchFixThreatsStatusUseCase(networkUtilsWrapper, scanStore, TEST_DISPATCHER)
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(true)
+    }
+
+    @Test
+    fun `given no network, when threats fix status is fetched, then NetworkUnavailable is returned`() = test {
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(false)
+
+        val useCaseResult = useCase.fetchFixThreatsStatus(fakeSiteId, listOf(fakeThreatId))
+            .toList(mutableListOf())
+            .last()
+
+        assertThat(useCaseResult).isEqualTo(Failure.NetworkUnavailable)
+    }
+
+    @Test
+    fun `when in progress threats fix status is fetched, then polling occurs until in progress status changes`() =
+        test {
+            whenever(scanStore.fetchFixThreatsStatus(any()))
+                .thenReturn(storeResultWithInProgressFixStatusModel)
+                .thenReturn(storeResultWithInProgressFixStatusModel)
+                .thenReturn(storeResultWithFixedFixStatusModel)
+
+            val useCaseResult = useCase.fetchFixThreatsStatus(fakeSiteId, listOf(fakeThreatId), delayInMs)
+                .toList(mutableListOf())
+
+            verify(scanStore, times(3)).fetchFixThreatsStatus(any())
+            assertThat(useCaseResult).containsSequence(InProgress, InProgress, Complete)
+        }
+
+    @Test
+    fun `given threats fixed successfully, when threats fix status is fetched, then Complete is returned`() = test {
+        whenever(scanStore.fetchFixThreatsStatus(any())).thenReturn(storeResultWithFixedFixStatusModel)
+
+        val useCaseResult = useCase.fetchFixThreatsStatus(fakeSiteId, listOf(fakeThreatId))
+            .toList(mutableListOf())
+            .last()
+
+        assertThat(useCaseResult).isEqualTo(Complete)
+    }
+
+    @Test
+    fun `given invalid response, when threats fix status is fetched, then RemoteRequestFailure is returned`() = test {
+        val storeResultWithGenericError = OnFixThreatsStatusFetched(
+            fakeSiteId,
+            FixThreatsStatusError(FixThreatsStatusErrorType.INVALID_RESPONSE),
+            FETCH_FIX_THREATS_STATUS
+        )
+        whenever(scanStore.fetchFixThreatsStatus(any())).thenReturn(storeResultWithGenericError)
+
+        val useCaseResult = useCase.fetchFixThreatsStatus(fakeSiteId, listOf(fakeThreatId))
+            .toList(mutableListOf())
+            .last()
+
+        assertThat(useCaseResult).isEqualTo(Failure.RemoteRequestFailure)
+    }
+
+    @Test
+    fun `when threats fix status is fetched, then FixFailure is returned if result model contains error`() = test {
```

Comment:
🔍 Nitpick: We might want to move `if result model contains error` into the "given" part -> `given result model contains error, when threats fix status is fetched, then FixFailure is returned`. Wdyt?

Reply:
In c6c955a